Sources/Common/RxManager.swift

```diff
@@ -6,20 +6,252 @@
+
+     - parameter keyPath: keyPath to look up json object to serialize. Ignore parameter or pass nil when json object is the json root item.
+
+     - returns: An instance of `Single<T>` filled with sample data specified on the RouteType.
+     */
+    func sampleObject<T: OperaDecodable>(_ route: RouteType, keyPath: String? = nil) -> Single<T> {
+        guard let json = JSONFrom(data: route.mockedData) else {
+            return Single.error(DecodingError.invalidMockedJson)
+        }
+        let object = keyPath.map({ (json as AnyObject).value(forKeyPath: $0) as Any}) ?? json
+        do {
+            let decodedData = try T.decode(object as Any)
+            return Single.just(decodedData)
+        } catch let error {
+            return Single.error(error)
+        }
+    }
+    
+    /**
+     Returns a `Single` of [T] for for the current RouteType. Notice that T conforms to OperaDecodable. If something goes wrong an `OperaSwift.Error` error is propagated through the result sequence.
+
+     - parameter collectionKeyPath: keyPath to look up json array to serialize. Ignore parameter or pass nil when json array is the json root item.
+
+     - returns: An instance of `Single<[T]>`  filled with sample data specified on the RouteType.
+     */
+    func sampleCollection<T: OperaDecodable>(_ route: RouteType, collectionKeyPath: String? = nil) -> Single<[T]> {
+        guard let json = JSONFrom(data: route.mockedData) else {
+            return Single.error(DecodingError.invalidMockedJson)
+        }
+        guard
+            let representation = (
+                collectionKeyPath.map {
+                    (json as AnyObject).value(forKeyPath: $0) as Any
+                    } ?? json
+                ) as? [[String: AnyObject]]
+            else {
+                return Single.error(DecodingError.invalidJson("Could not find keypath on the decoded json"))
+        }
+
+        var result = [T]()
+        representation.forEach {
+            if let decodedData = try? T.decode($0 as AnyObject) {
+                result.append(decodedData)
+            }
+        }
+        return Single.just(result)
+    }
+
+    /**
+     Returns a `Single` sequence for the current request using the specified mocks. If something goes wrong an `OperaSwift.Error` error is propagated through the result sequence.
+
+     - returns: An instance of `Single<Any>`
+     */
+    func sampleAny(_ route: RouteType) -> Single<Any> {
+        guard let json = JSONFrom(data: route.mockedData) else {
+            return Single.error(DecodingError.invalidMockedJson)
+        }
+        return Single.just(json as Any)
+    }
+
+    /**
+     Returns an empty `Completable` sequence for the current request. No networking call is done as this is supposed to be used as part of a mocked response
+
+     - returns: An instance of `Completable`
+     */
+    func sampleCompletableResponse(_ route: RouteType) -> Completable {
+        return Completable.empty()
```

Comment:
Does this work? If it is like Observable.empty() this will never emit to the subscriber